### PR TITLE
Chore: Improve TypeScript, Linting, and Styling Support in Doc MDX Files

### PR DESCRIPTION
**Pull request by moathabuhamad-cengage**
Issue: #1553 

## What I did
   - Updated ESLint configuration `.eslintrc.js` to include `@typescript-eslint/parser` and `mdx` plugins.
   - Added rules to handle MDX files and TypeScript.
   - Disabled specific rules for MDX files to avoid unnecessary warnings.
   - Added `lint`, `lint:fix`, and `lint:check` scripts for consistent linting.
   - Updated `format` script to include more file type

## Screenshots:
Screenshot from 2024-11-22 15-03-38

## Checklist 
- [x] changeset has been added
- [x] Pull request description is descriptive
- [x] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes
- [-] I have added tests that prove my fix is effective or that my feature works

## How to test
- Run linting and formatting scripts
- Ensure MDX files are properly linted and formatted.
- Check for any linting or formatting errors.

### CODE REVIEW

File: website/react-magma-docs/src/pages/api/alert.mdx [hunk 1]
```diff
@@ -25,9 +24,9 @@ import { Alert } from 'react-magma-dom';
 export function Example() {
   return (
     <>
-      <Alert isDismissible>I'm an Alert</Alert>
+      <Alert isDismissible>Im an Alert</Alert>
       <Alert>
-        I'm an Alert with <a href="#">linked text</a>
+        Im an Alert with <a href="#">linked text</a>
```
Comment (silvalaura) [inline website/react-magma-docs/src/pages/api/alert.mdx:29]: I'm

**Review by silvalaura — COMMENTED** [review]
Haven't had a chance to navigate through every single page, but code wise this looks good!

File: website/react-magma-docs/src/pages/api/card.mdx [hunk 2]
```diff
@@ -199,7 +197,7 @@ export function Example() {
         height="125"
         width="250"
         src="https://placekitten.com/250/125"
-        alt="Image of a cat"
+        alt="cat"
```
[the remark this answers is not in the document]
Comment (moathabuhamad-cengage) [reply website/react-magma-docs/src/pages/api/card.mdx:200]: breaks eslint rule [jsx-a11y/img-redundant-alt](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/tree/HEAD/docs/rules/img-redundant-alt.md)

File: website/react-magma-docs/src/pages/api/combobox.mdx [hunk 3]
```diff
@@ -185,25 +183,19 @@ export function Example() {
...
       case ComboboxStateChangeTypes.InputKeyDownEscape:
       case ComboboxStateChangeTypes.InputBlur:
         updateIsOpen(false);
-        updateHighlightedIndex(newChanges.highlightedIndex);
-        break;
-      case ComboboxStateChangeTypes.ItemClick:
-      case ComboboxStateChangeTypes.InputKeyDownEnter:
-        updateHighlightedIndex(newChanges.highlightedIndex);
-        updateIsOpen(newChanges.isOpen);
-        updateSelectedItem(newChanges.selectedItem);
```
[the remark this answers is not in the document]
Comment (moathabuhamad-cengage) [reply website/react-magma-docs/src/pages/api/combobox.mdx]: fixed

File: website/react-magma-docs/src/pages/design/stepper.mdx [hunk 4]
```diff
@@ -252,41 +266,53 @@ If the user cannot proceed due to a server-side issue, then an inline alert shou
...
-  <div class="image-container">
-    <img src="../../images/stepper/mobile-standard-labels.png" alt="GATSBY_EMPTY_ALT" />
+  <div className="image-container">
+    <img
+      src="../../images/stepper/mobile-standard-labels.png"
+      alt="GATSBY_EMPTY_ALT"
+    />
   </div>
-</figure> 
+</figure>{' '}
```
Comment (silvalaura) [inline website/react-magma-docs/src/pages/design/stepper.mdx:280]: image

File: website/react-magma-docs/src/pages/design/stepper.mdx [hunk 5]
```diff
@@ -252,41 +266,53 @@ If the user cannot proceed due to a server-side issue, then an inline alert shou
...
-  <div class="image-container">
-    <img src="../../images/stepper/mobile-summary-view.png" alt="GATSBY_EMPTY_ALT" />
+  <div className="image-container">
+    <img
+      src="../../images/stepper/mobile-summary-view.png"
+      alt="GATSBY_EMPTY_ALT"
+    />
   </div>
-</figure>  
+</figure>{' '}
```
Comment (silvalaura) [inline website/react-magma-docs/src/pages/design/stepper.mdx:306]: image

File: website/react-magma-docs/src/pages/api-intro/styles-and-themes.mdx
Comment (silvalaura) [inline website/react-magma-docs/src/pages/api-intro/styles-and-themes.mdx:1]: image

File: website/react-magma-docs/src/pages/api/appbar.mdx
Comment (silvalaura) [inline website/react-magma-docs/src/pages/api/appbar.mdx:1]: Not part of this work necessarily, but I just noticed that `# AppBar Props` should be `## AppBar Props` so it appears on the secondary menu

File: website/react-magma-docs/src/pages/api/buttongroup.mdx
Comment (silvalaura) [inline website/react-magma-docs/src/pages/api/buttongroup.mdx:1]: Same with `# ButtonGroup Props` => `## ButtonGroup Props`

File: website/react-magma-docs/src/pages/api/character-counter.mdx
Comment (silvalaura) [inline website/react-magma-docs/src/pages/api/character-counter.mdx:1]: `### Character Counter Props` => `## Character Counter Props`

File: website/react-magma-docs/src/pages/api/dropzone.mdx
Comment (silvalaura) [inline website/react-magma-docs/src/pages/api/dropzone.mdx:1]: Can we change `## File Uploader Props` => `## Dropzone Props`

File: website/react-magma-docs/src/pages/api/grid.mdx
Comment (silvalaura) [inline website/react-magma-docs/src/pages/api/grid.mdx:1]: `# Grid Props` => `## Grid Props`
&
`# Grid Items Props` => `## Grid Items Props`

File: website/react-magma-docs/src/pages/api/icon.mdx
Comment (silvalaura) [inline website/react-magma-docs/src/pages/api/icon.mdx:1]: image

File: website/react-magma-docs/src/pages/api/spacer.mdx
Comment (silvalaura) [inline website/react-magma-docs/src/pages/api/spacer.mdx:1]: `# Spacer Props` => `## Spacer Props`

File: website/react-magma-docs/src/pages/api/tooltip.mdx
Comment (silvalaura) [inline website/react-magma-docs/src/pages/api/tooltip.mdx:1]: image

File: website/react-magma-docs/src/pages/api/transition.mdx
Comment (silvalaura) [inline website/react-magma-docs/src/pages/api/transition.mdx:1]: `# Transition Props` => `## Transition Props`

**Review by silvalaura — CHANGES_REQUESTED** [review]

**Pull request merged**